app/models/post.rb

```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  before_destroy :decrement_user_posts_counter
+  after_save :increment_user_posts_counter
```

Comment:
- [ ] Great work creating a method that reduces the `posts_counter` anytime a post is destroyed. It is really impressive. 👏